Update docs for ARE_ANY_VEHICLE_SEATS_FREE and SET_VEHICLE_UNDRIVEABLE (#1324)
citizenfx/natives: VEHICLE/AreAnyVehicleSeatsFree.md

BOOL ARE_ANY_VEHICLE_SEATS_FREE(Vehicle vehicle);
```

Dead peds still count as occupying a seat until the body is removed, so a vehicle full of corpses returns `false`. Peds tasked to enter the vehicle but not yet inserted do not occupy their target seat.

> For per-seat queries use [IS_VEHICLE_SEAT_FREE](#_0x22AC59A870E6A669), and to resolve which ped holds a specific seat use [GET_PED_IN_VEHICLE_SEAT](#_0xBB40BBB9B9A067B).

Reviewer: ```suggestion
For per-seat queries use [IS_VEHICLE_SEAT_FREE](#_0x22AC59A870E6A669), to find which ped is in a specific seat use [GET_PED_IN_VEHICLE_SEAT](#_0xBB40BBB9B9A067B).
```

Author: applied

BOOL ARE_ANY_VEHICLE_SEATS_FREE(Vehicle vehicle);
```

> Returns `true` when the vehicle has at least one seat with no ped assigned to it. Vehicles with no seats in their model (trailers, prop carriages) and invalid handles both return `false`.

Reviewer: ```suggestion
```

This is just duplicating the return section

Author: Applied

> Returns `true` when the vehicle has at least one seat with no ped assigned to it, and `false` when every seat is filled. The check covers every seat in the vehicle's model layout, including the driver seat, so a bike or single-seat plane returns `true` only when that one seat is empty. An invalid vehicle handle returns `false`, and vehicles whose model defines zero seats (trailers, prop carriages) always return `false` because there are no seats to be free.

Reviewer: ```suggestion
```
This seems *really* dense, and most of this information could probably be assumed given the natives name.

Author: Will simplify but retain two critical piece of info:

- Zero-seat vehicles like trailers return false.
- Invalid handle returns false.

Dead peds still count as occupying a seat until the body is removed, so a vehicle full of corpses returns `false`. Peds tasked to enter the vehicle but not yet inserted do not occupy their target seat.

For per-seat queries use [IS_VEHICLE_SEAT_FREE](#_0x22AC59A870E6A669), and to resolve which ped holds a specific seat use [GET_PED_IN_VEHICLE_SEAT](#_0xBB40BBB9B9A067B).

## Parameters
* **vehicle**: The vehicle handle to check.

## Return value

> Returns `true` if any seat on the vehicle is unoccupied, `false` if every seat is filled or the handle is not a valid vehicle.

Reviewer: ```suggestion
Returns `true` when the vehicle has at least one seat with no ped assigned to it. Vehicles with no seats in their model (trailers, prop carriages) or an invalid entity will return `false`.
```

Author: Applied

> A seat only counts as occupied while a ped pointer is actually stored in it. Dead peds still count as occupying their seat until the body is removed, so a vehicle full of corpses will read as full. Peds that have been scheduled to enter the vehicle but have not yet been inserted do not occupy their target seat, so those seats keep reading as free until the ped is placed.

Reviewer: ```suggestion
Dead peds still count as occupying a seat until the body is removed, so a vehicle full of corpses will still return `true`. 

Peds that have been tasked to enter a vehicle but have not yet gotten in do not occupy their target seat.
```

Author: Will simplify, however, a vehicle full of dead corpses will return false.

> The native does not consider lock state, door damage, shuffle gating, or whether the caller could jack an existing occupant. Scripts that need "empty and enterable" semantics should pair this with the relevant lock and accessibility checks. For per-seat queries use [IS_VEHICLE_SEAT_FREE](#_0x22AC59A870E6A669), and to resolve which ped holds a specific seat use [GET_PED_IN_VEHICLE_SEAT](#_0xBB40BBB9B9A067B).

Reviewer: ```suggestion
```

I don't see how this section is necessary for this native.

> `true` if any seat on the vehicle is unoccupied, `false` if every seat is filled or the handle is not a valid vehicle.

Reviewer: ```suggestion
Returns `true` if any seat on the vehicle is unoccupied, `false` if every seat is filled or the handle is not a valid vehicle.
```

Author: Will apply
